download_action.go

```diff
@@ -100,5 +102,9 @@ func (d *DownloadAction) Run(context *DebosContext) error {
 		return fmt.Errorf("Unsupported URL is provided: '%s'", url.String())
 	}
 
+	if d.Unpack == true {
+		return UnpackTarArchive(filename, context.rootdir, d.Compression)
```

Comment:
Download actions shouldn't unpack to the root directory. Both the download and the unpack should go into a seperate area such that follow-up actions can pick an choice what to put where rather then mandating a tarball following your root layout.

So the general idea is: 
* Download action, give the result some tag name.
* Overlay, copy, raw actions to transfer the result into the rootfs/image/whatever (this can be done N times).

e.g. one example could be downloading a tarball with firmware and then copying that firmware to sepcific location on the rootfs.

Fwiw the source field in the raw action is already going towards this, by having filesystem as the currently supported source (e.g. dd from something inside the filesystem we build)